base/dcp_common.go

```diff
@@ -522,15 +522,13 @@ func dcpKeyFilter(key []byte, metaKeys *MetadataKeys) bool {
 }
 
 // Makes a feedEvent that can be passed to a FeedEventCallbackFunc implementation
+// The byte slices must be copied to ensure that memory associated with the memd mutationEvent and Packet are independent and can be released or reused by gocbcore as needed.
 func makeFeedEvent(key []byte, value []byte, dataType uint8, cas uint64, expiry uint32, vbNo uint16, collectionID uint32, opcode sgbucket.FeedOpcode) sgbucket.FeedEvent {
 
-	// not currently doing rq.Extras handling (as in gocouchbase/upr_feed, makeUprEvent) as SG doesn't use
-	// expiry/flags information, and snapshot handling is done by cbdatasource and sent as
-	// SnapshotStart, SnapshotEnd
 	event := sgbucket.FeedEvent{
 		Opcode:       opcode,
-		Key:          key,
-		Value:        value,
+		Key:          EfficientBytesClone(key),
```

Comment:
This cloning can be removed now that the incoming key/value are already copied, correct?

Reply:
This codepath is the one still used via CBGT (Invoked via `DCPDest`->`DataUpdate`->`makeFeedEventForDest`) for EE/Sharded import.

The other ones that changed with `mutationEvent`/`deletionEvent` are for the caching feed via `StartGocbDCPFeed`->`DCPClient`->`Mutation`